hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/AvroMessageContentWrapper.java

```diff
@@ -45,11 +45,12 @@ UnwrappedMessageContent unwrapContent(byte[] data, CompiledSchema<Schema> schema
     private MessageMetadata getMetadata(Map<Utf8, Utf8> metadata) {
         if (metadata == null) {
             long timestamp = clock.millis();
-            return new MessageMetadata(timestamp, MessageId.forTimestamp(timestamp), Collections.EMPTY_MAP);
+            return new MessageMetadata(timestamp, MessageId.empty(), Collections.EMPTY_MAP);
```

Comment:
Maybe skip the `MessageId` param in that constructor and have `MessageMetadata` handle the case of missing ID - we get single responsibility for free.